src/gui/guiFormSpecMenu.cpp

```diff
@@ -2695,6 +2696,84 @@ void GUIFormSpecMenu::parseSetFocus(const std::string &element)
 		<< "'" << std::endl;
 }
 
+void GUIFormSpecMenu::parseModel(parserData *data, const std::string &element)
+{
+	std::vector<std::string> parts = split(element, ';');
+
+	if (parts.size() < 5 || (parts.size() > 8 &&
+			m_formspec_version <= FORMSPEC_API_VERSION)) {
+		errorstream << "Invalid model element (" << parts.size() << "): '" << element
+			<< "'" << std::endl;
+		return;
+	}
+
+	parts.resize(std::max((s32)parts.size(), 7));
```

Comment:
I believe this is the source of the segfault.  `model[]` can have up to 8 parameters, this resizes it to only 7.